main.tf

```diff
@@ -1,8 +1,35 @@
-# TODO: Replace this dummy resource azurerm_resource_group.TODO with your module resource
-resource "azurerm_resource_group" "TODO" {
-  location = var.location
-  name     = var.name # calling code must supply the name
-  tags     = var.tags
+data "azurerm_resource_group" "rg" {
+  name = var.resource_group_name
+}
+
+resource "azapi_resource" "eventgrid_namespace" {
```

Comment:
Please set `response_export_values` explicitly to avoid plan noise. Only output the the items required.